notebooks/stress-tests.ipynb

```diff
@@ -0,0 +1,554 @@
+      "   47       200    0.057s    0.401s\n",
+      "   48       200    0.013s    0.385s\n",
+      "   49       200    0.067s    0.402s\n",
+      "   50       200    0.088s    0.447s\n",
+      "   51       200    0.089s    0.401s\n",
+      "   52       200    0.055s    0.476s\n",
+      "   53       200    0.024s    1.036s\n",
+      "   54       200    0.017s    0.876s\n",
+      "   55       200    0.023s    0.603s\n",
+      "   56       200    0.099s    0.376s\n",
+      "   57       200    0.022s    0.461s\n",
+      "   58       200    0.057s    0.409s\n",
+      "   59       200    0.071s    0.561s\n",
+      "   60       200    0.076s    0.371s\n",
+      "   61       200    0.067s    0.389s\n",
+      "   62       200    0.030s    0.381s\n",
+      "   63       200    0.085s    0.413s\n",
+      "   64       200    0.049s    0.428s\n",
+      "   65       200    0.068s    0.472s\n",
+      "   66       200    0.011s    0.917s\n",
+      "   67       200    0.006s    0.906s\n",
+      "   68       200    0.092s    0.756s\n",
+      "   69       200    0.072s    0.418s\n",
+      "   70       200    0.092s    0.399s\n",
+      "   71       200    0.091s    0.387s\n",
+      "   72       200    0.065s    0.394s\n",
+      "   73       200    0.052s    0.391s\n",
+      "   74       200    0.054s    0.455s\n",
+      "   75       200    0.097s    0.356s\n",
+      "   76       200    0.027s    0.375s\n",
+      "   77       200    0.066s    0.365s\n",
+      "   78       200    0.061s    0.405s\n",
+      "   79       200    0.051s    0.632s\n",
+      "   80       200    0.011s    0.752s\n",
+      "   81       200    0.007s    0.741s\n",
+      "   82       200    0.033s    0.562s\n",
+      "   83       200    0.008s    0.386s\n",
+      "   84       200    0.087s    0.382s\n",
+      "   85       200    0.065s    0.384s\n",
+      "   86       200    0.088s    0.366s\n",
+      "   87       200    0.040s    0.399s\n",
+      "   88       200    0.092s    0.398s\n",
+      "   89       200    0.085s    0.364s\n",
+      "   90       200    0.011s    0.403s\n",
+      "   91       200    0.078s    0.373s\n",
+      "   92       200    0.017s    0.398s\n",
+      "   93       200    0.029s    0.484s\n",
+      "   94       200    0.044s    0.904s\n",
+      "   95       200    0.039s    0.862s\n",
+      "   96       200    0.012s    0.498s\n",
+      "   97       200    0.096s    0.407s\n",
+      "   98       200    0.058s    0.374s\n",
+      "   99       200    0.036s    0.372s\n",
+      "Times:\n",
+      "  min: 0.349s\n",
+      "  avg: 0.471s\n",
+      "  max: 1.036s\n"
+     ]
+    }
+   ],
+   "source": [
+    "# NBVAL_IGNORE_OUTPUT\n",
+    "\n",
+    "for bird in [\"finch\", \"flyingpigeon\", \"raven\"]:\n",
+    "    bird_url = f\"{TWITCHER_URL}/{bird}/wps?service=wps&request=getcapabilities\"\n",
+    "    assert stress_test_requests(bird_url, runs=100, max_avg_time=1) == 0, \"Failure condition encountered\""
```

Comment:
Could we expand this to something like below for better reporting?  This runs failed https://daccs-jenkins.crim.ca/job/PAVICS-e2e-workflow-tests/job/stress-test/1/console and it's impossible to determine the root cause from Jenkins console output.
```
stress_test_result = stress_test_requests(bird_url, runs=100, max_avg_time=1)

if (stress_test_result.result == 1):
    # example "Failed: some non 200 response code observed within the total 100 runs: 401, 401, 401, 409, 500, 502"
    raise AssertionError(f"Failed: some non {stress_test_result.code} response code observed within the total {stress_test_result.runs} runs: {stress_test_result.bad_errors_codes}")

if (stress_test_result.result == 2):
    # example: "Failed: avg_time (1.034s) > max_avg_time (1s)"
    raise AssertionError("Failed: avg_time ({stress_test_result.avg_time}s) > max_avg_time ({stress_test_result.max_avg_time}s)")
```

This implies changing the function `stress_test_requests` to return a dataclass instead, which I think is not too much work since all the info are already available, just have to shuffle them into the dataclass.

Reply:
@tlvu 
You can see in the latest commit. Updated to use dataclass and reporting the error/results as suggested.